### PR TITLE
feature: Add Squid application support

### PR DESCRIPTION
**Pull request by VVelox**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

This requires https://github.com/librenms/librenms-agent/pull/93 .

Also this seems to have slightly got manangled with some of the white cleanup I've been doing.

### CODE REVIEW

**Comment by murrant** [comment]
@VVelox You have a lot of old commits on here, that isn't specifically an issue but can cause headaches for yourself.  Are you using branches to develop?
http://docs.librenms.org/Developing/Using-Git/

**Comment by VVelox** [comment]
Yeah, using branches. Unfortunately I seem to of branched the Squid stuff of of my white cleanup instead of master.

As to the stuff that has already been merged, not sure why that is coming up. I've been unable to find any decent answer about that.

**Comment by VVelox** [comment]
Any ways, will get the white cleanup stuff cleaned out of this branch. I can't seem to figure out how to get ride of the all the stuff that has already been merged though.

**Comment by VVelox** [comment]
Okay. Stuff that got included in this given improper branching has been cleaned up.

**Comment by VVelox** [comment]
@murrant as far as I can tell showing all the commits to my clone of this repo since it's inception is the default behavior and made a bit worse in this case by the fact I did the fail2ban and php-fpm stuff in the master of my clone when I initially got started instead of a branch like I've done with stuff since then.

File: includes/polling/applications/squid.inc.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+$name = 'squid';
+$app_id = $app['app_id'];
+$options      = '-O qv';
+$mib          = 'NET-SNMP-EXTEND-MIB';
+$oid          = 'nsExtendOutLine.5.115.113.117.105.100';
+$returned = snmp_walk($device, $oid, $options, $mib);
+
+# cacheCurrentLRUExpiration is returned here, but we don't use it as it would be more work to exclude it from the extend
+list( $MemMaxSize, $SwapMaxSize, $SwapHighWM, $SwapLowWM, $SysPageFaults, $SysNumReads, $MemUsage, $CpuTime, $CpuUsage,
```
Comment (laf) [inline includes/polling/applications/squid.inc.php]: Space after `list( ` and also at the end can be removed.

**Comment by VVelox** [comment]
Removed the need for the extend. Updated the docs and will begin working on updating the poller later.

**Comment by laf** [comment]
Looks good, if you can drop the require 'includes/graphs/common.inc.php'; lines + rebase that would be ace.

**Comment by VVelox** [comment]
Done. :)

**Comment by laf** [comment]
Sorry also lowercase the variables as well :(

**Comment by VVelox** [comment]
All lower cased and conflicted merged.

**Comment by laf** [comment]
Scrut fix committed, should be good to merge once travis is happy

**Pull request merged**